### ISSUE TITLE
linter: `anchor-is-valid` rule of `jsx-11y` ignores the attributes setting of the plugin

### ISSUE THREAD

**Issue by SkandarS0** [oxc-project/oxc#20552]
### What version of Oxlint are you using?

1.56.0

### What command did you run?

oxlint 

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "../../node_modules/oxlint/configuration_schema.json",
  "plugins": ["jsx-a11y"],
  "settings": {
    "jsx-a11y": {
      "components": {
        "Link": "a",
        "Button": "button",
        "Form": "form",
      },
      "attributes": {
        "to": ["href"],
      },
    },
    "react": {
      "linkComponents": [
        {
          "name": "Link",
          "attribute": "to",
        },
      ],
      "version": "19.2.4",
    },
  },
}

```

### What happened?

running oxlint should have skipped the `Link` components with the `to` attribute, but it emits an error/warning.
Though if there is a spread operator, it doesn't emit an error/warning, so it acknowledges the `components ` setting.
```
  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Missing `href` attribute for the `a` element.
    ╭─[file:///D:/projects/f-med-sousse/apps/web/src/features/auth/ui/login-form.tsx:70:14]
 69 │           <Button variant="link" size="sm">
 70 │             <Link to="/forgot-password" preload="intent">
    ·              ────
 71 │               {t("pages:login.components.form.forgotPassword")}
    ╰────
  help: Provide an `href` for the `a` element.
```
